config/config.go

```diff
@@ -526,6 +526,8 @@ type HTTPClientConfig struct {
 	ProxyURL URL `yaml:"proxy_url,omitempty"`
 	// TLSConfig to use to connect to the targets.
 	TLSConfig TLSConfig `yaml:"tls_config,omitempty"`
+	// If set, override whether to use HTTP KeepAlive - scraping defaults OFF, remote read/write defaults ON
```

Comment:
I think this is going to be very hard to understand for users and to explain to them, a given config stanza shouldn't mean different things in different contexts. It's also unresolved what scraping should do in relation to keepalive.

If you want to enable keepalive for remote storage you should do exactly that, without bringing additional complexity and other debates into it.

Reply:
I'm not following why you think this config means different things in different contexts.  `false` always means you want keep-alive disabled, and `true` always means you want it available.  Unless I made a typo?

Reply:
No config means different things in different contexts. We don't do this anywhere else in Prometheus or the other components.

Reply:
I made a different take on it at #3173, with no configurability.
Let me know if you revert this PR.